.github/workflows/run-build.yml

```diff
@@ -2,6 +2,10 @@ name: run-build
 on:
   pull_request: {}
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
```

Comment:
For PR related runs, we might as well cancel the previously running action so we can get our most recent results as quickly as possible (and save on run minutes)